py/server/deephaven/experimental/sql.py

```diff
@@ -0,0 +1,64 @@
+#
+# Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+#
+import contextlib
+import inspect
+import jpy
+from typing import Dict, Union, Mapping, Optional, Any
+
+from deephaven import DHError
+from deephaven.table import Table, _j_py_script_session
+
+_JSql = jpy.get_type("io.deephaven.engine.sql.Sql")
+
+
+@contextlib.contextmanager
+def _scope(globals: Mapping[str, Any], locals: Mapping[str, Any]):
+    j_py_script_session = _j_py_script_session()
+    # Can do `globals | locals` in Python 3.9+
+    j_py_script_session.pushScope({**globals, **locals})
+    try:
+        yield
+    finally:
+        j_py_script_session.popScope()
+
+
+def eval(
+    sql: str,
+    dry_run: bool = False,
+    globals: Optional[Mapping[str, Any]] = None,
+    locals: Optional[Mapping[str, Any]] = None,
+) -> Union[Table, jpy.JType]:
+    """Experimental SQL evaluation. Subject to change.
+
+    If both globals and locals is omitted (the default), the sql is executed with the globals and locals in the
+    environment where eval() is called.
```

Comment:
```suggestion
    """Evaluates an SQL query and returns the result.
    
    Variables in the query string are taken from the environment when this method is called.
```
Reworded to be more consistent with our other docs.
The experimental package is understood to be experimental and subject to change, so no need to repeat that here.

Reply:
Right now, it's only named tables that we can use. There may be potential in the future to use constants or other variables as input to SQL, but that hasn't been integrated yet.